lib/scholar/neighbors/radius_nearest_neighbors.ex

```diff
@@ -0,0 +1,347 @@
+defmodule Scholar.Neighbors.RadiusNearestNeighbors do
+  @moduledoc """
+  The K-Nearest Neighbors.
+  """
+  import Nx.Defn
+  import Scholar.Shared
+  require Nx
+
+  @derive {Nx.Container,
+           keep: [:weights, :num_classes, :task, :metric, :radius], containers: [:data, :labels]}
+  defstruct [:data, :labels, :weights, :num_classes, :task, :metric, :radius]
+
+  opts = [
+    radius: [
+      type: {:custom, Scholar.Options, :positive_number, []},
+      default: 1.0,
+      doc: "Radius of neighborhood"
+    ],
+    num_classes: [
+      type: :pos_integer,
+      doc: "Number of classes in provided labels"
+    ],
+    weights: [
+      type: {:in, [:uniform, :distance]},
+      default: :uniform,
+      doc: """
+      Weight function used in prediction. Possible values:
+
+        * `:uniform` - uniform weights. All points in each neighborhood are weighted equally.
+
+        * `:distance` - weight points by the inverse of their distance. in this case, closer neighbors of
+          a query point will have a greater influence than neighbors which are further away.
+      """
+    ],
+    metric: [
+      type: {:custom, Scholar.Options, :metric, []},
+      default: {:minkowski, 2},
+      doc: ~S"""
+      Name of the metric. Possible values:
+
+      * `{:minkowski, p}` - Minkowski metric. By changing value of `p` parameter (a positive number or :infinity)
+        we can set Manhattan (1), Euclidean (2), Chebyshev (:infinity), or any arbitrary $L_p$ metric.
+
+      * `:cosine` - Cosine metric.
+      """
+    ],
+    task: [
+      type: {:in, [:classification, :regression]},
+      default: :classification,
+      doc: """
+      Task that will be performed using K Nearest Neighbors. Possible values:
+
+      * `:classification` - Classifier implementing the Radius Nearest Neighbors vote.
+
+      * `:regression` - Regression based on Radius Nearest Neighbors.
+        The target is predicted by local interpolation of the targets associated of the nearest neighbors in the training set.
+      """
+    ]
+  ]
+
+  @opts_schema NimbleOptions.new!(opts)
+
+  @doc """
+  Fit the K-nearest neighbors classifier from the training data set.
+
+  For classification, provided labels needs to be consecutive natural numbers. If your labels does
```

Comment:
```suggestion
  For classification, provided labels need to be consecutive positive integers. If your labels does
```

"natural numbers" is unfortunately an ambiguous definition. I assumed you don't want to include zero. If you do, use non-negative instead of positive there.

Reply:
Yes, in this case I mean non-negative.

Reply:
```suggestion
  For classification, provided labels need to be consecutive non-negative integers. If your labels does
```